examples/convert_nirs_to_bids.py

```diff
@@ -0,0 +1,240 @@
+"""
+====================================
+13. Convert NIRS data to BIDS format
+====================================
+
+In this example, we use MNE-BIDS to create a BIDS-compatible directory of NIRS
+data. Specifically, we will follow these steps:
+
+1. Download some NIRS data from a github repository
+
+2. Load the data, extract information, and save it in a new BIDS directory.
+
+3. Check the result and compare it with the standard.
+
+4. Cite ``mne-bids``.
+
+.. currentmodule:: mne_bids
+
+.. _BrainVision format: https://www.brainproducts.com/productdetails.php?id=21&tab=5
+.. _CapTrak: https://www.fieldtriptoolbox.org/faq/coordsys/#details-of-the-captrak-coordinate-system
+
+"""  # noqa: E501
+
+# Authors: Robert Luke <code@robertluke.net>
+#
+# License: BSD-3-Clause
+
+# %%
+# We are importing everything we need for this example:
+import os.path as op
+import shutil
+import urllib.request
+import tempfile
+
+import mne
+
+from mne_bids import write_raw_bids, BIDSPath, print_dir_tree
+from mne_bids.stats import count_events
+
+# %%
+# Download the data
+# -----------------
+#
+# First, we need some data to work with. We will use the
+# `Finger Tapping Dataset <https://github.com/rob-luke/BIDS-NIRS-Tapping>`_
+# available on GitHub.
+#
+# The data consists of 109 volunteers performing 14 experimental runs each.
+# For each subject, there were two baseline tasks (i) eyes open, (ii) eyes
+# closed, as well as four different motor imagery tasks.
+#
+# In this example, we will download the data for a single subject doing the
+# baseline task "eyes closed" and format it to the Brain Imaging Data Structure
+# (`BIDS <https://bids.neuroimaging.io/>`_).
+#
+
+# Create a temporary directory to store the data in
+data_dir = tempfile.mkdtemp()
+file_path = f"{data_dir}/sub-01_task-tapping_nirs.snirf"
+
+# Download the data for subject 1.
+urllib.request.urlretrieve("https://github.com/rob-luke/BIDS-NIRS-Tapping/raw"
+                           "/01-Raw-to-SNIRF/sub-01/nirs/"
+                           "sub-01_task-tapping_nirs.snirf",
+                           file_path)
```

Comment:
Is there a cleaner way to do this? I don't know if this will work on all operating systems

Reply:
MNE now officially depends on Pooch, so you could use that for downloading arbitrary files from arbitrary locations!

Check out this example here where they basically do what you want to do: download a file from GitHub:

https://www.fatiando.org/pooch/latest/retrieve.html#retrieve

Reply:
I've implemented this now.